kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaProxyContext.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.time.Clock;
+import java.util.stream.Collectors;
+
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.VirtualKafkaCluster;
+import io.kroxylicious.kubernetes.operator.model.ProxyModel;
+import io.kroxylicious.kubernetes.operator.model.ProxyModelBuilder;
+
+public class KafkaProxyContext {
+
+    private static final String KEY_SEC = "sec";
+    private static final String KEY_MODEL = "model";
+    private static final String KEY_CLOCK = "clock";
+
+    private KafkaProxyContext() {
+    }
+
+    static void init(Clock clock,
+                     SecureConfigInterpolator secureConfigInterpolator,
+                     KafkaProxy proxy,
+                     Context<KafkaProxy> context) {
+        var rc = context.managedWorkflowAndDependentResourceContext();
+
+        rc.put(KEY_CLOCK, Clock.fixed(clock.instant(), clock.getZone()));
```

Comment:
maybe it would be better to stash a `KafkaProxyContext` record in the resource context, we have 3 mandatory fields with some boilerplate per type to push it in and out of the context, we could make them fields of a KPC, and make `isBroken(VirtualKafkaCluster cluster)` a method of KPC.

Reply:
Doing that wouldn't encapsulate the map key though.

Reply:
you can still hide the map key right? with static methods to access a KPC record that's stored in the context?